src/natcap/invest/managed_raster/ManagedRaster.h

```diff
@@ -337,15 +340,15 @@ class ManagedRaster {
         }
 };
 
+template<class T>
 class ManagedFlowDirRaster: public ManagedRaster {
-
 public:
-
     ManagedFlowDirRaster() {}
 
     ManagedFlowDirRaster(char* raster_path, int band_id, bool write_mode)
         : ManagedRaster(raster_path, band_id, write_mode) {}
 
+    template<typename T_ = T, std::enable_if_t<std::is_same<T_, MFD>::value>* = nullptr>
```

Comment:
This tells the compiler to use this implementation of `is_local_high_point` if the template argument is `MFD`. The same pattern is used in all the iterator classes below.

Reply:
Interesting!  So then if we were to extend these classes to accommodate some additional routing method, we would need to add new classmethods that have a conditional like this?